Software/Development/SemesterProject/App/controlTask.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * controlTask.c
+ *
+ *  Created on: 22. mar. 2019
+ *      Author: Nicolai Fransen
+ */
+
+#include "controlTask.h"
+
+/*
+ * This function will check the state of the open-loop/closed-loop switch,
+ * and from that execute the desired operation mode.
+ */
+void executeControl(void)
+{
+    if (isOpenLoopControlSelected())

```

Comment:
as discussed, I think that the system manager state should be checked here. It seems a simpler system.

So the idea would be:
if (sysMgrState = Running)  {    do_control()}
else error_reaction();

error_reaction()
{
    set_duties_to_0();
    set_driver_enable_to_0();
    restart_control_vars();//here or downstream, just think about it
    //more error reactions?
}

Reply:
This have been changed now